app/validators/previously_published_validator.rb

```diff
@@ -0,0 +1,16 @@
+class PreviouslyPublishedValidator < ActiveModel::Validator
+  def validate(record)
+    record.has_previously_published_error = false
+
+    if record.previously_published.nil?
+      record.errors[:base] << "You must specify whether the document has been published before"
+      record.has_previously_published_error = true
+    elsif record.previously_published && record.first_published_at.blank?
+      record.errors.add(:first_published_at, "can't be blank")
```

Comment:
Should this branch (and the `if` below) also set `has_previously_published_error`?  The `has_first_published_error` logic that this replaces does that: https://github.com/alphagov/whitehall/pull/4076/files#diff-d9c2d399fb6d507d2c0c4da96feece4eL647

Reply:
The reason `has_previously_published_error` exists is to apply some frontend formatting when the user hasn't selected any option in the radio buttons - https://github.com/alphagov/whitehall/blob/3adb1d2b560e2d77bf560b7b22cef10793a296be/app/views/admin/editions/_first_published_at.html.erb#L2
I think the reason this is needed is because `previously_published` is a transient attribute, whereas`first_published_at` isn't, so when there is an error on it the generic bootstrap red formatting will be applied as per:
<img width="365" alt="screen shot 2018-06-14 at 14 15 54" src="https://user-images.githubusercontent.com/8225167/41414272-81dcc252-6fdd-11e8-8395-6f762e6a8254.png">